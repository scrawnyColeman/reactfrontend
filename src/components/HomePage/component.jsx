import React from 'react';
import styled from 'styled-components';
import PopularCourses from '../PopularCourses/component';
import UserCourses from '../UserCourses/component';
import data from '../../data/courses.js';

const StyledItemWrapper = styled.div`
    display: grid;
    grid-template-columns: 3fr 2fr;
    margin: 20vh 20px 0 20px;
`;

const HomePage = () => {
    const { courses } = data;
    return (
        <StyledItemWrapper>
            <UserCourses courses={courses} />
            <PopularCourses courses={courses} />
        </StyledItemWrapper>
    );
};
export default HomePage;
