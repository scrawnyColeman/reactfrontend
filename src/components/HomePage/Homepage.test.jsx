import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import HomePage from './component';

Enzyme.configure({ adapter: new Adapter() });

describe('HomePage component', () => {
    test('component matches snapshot', () => {
        const wrapper = shallow(<HomePage />);
        expect(wrapper).toMatchSnapshot();
    });
});
