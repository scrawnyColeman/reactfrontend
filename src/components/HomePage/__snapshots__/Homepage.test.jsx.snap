// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component component matches snapshot 1`] = `
<styled.div>
  <UserCourses
    courses={
      Array [
        Object {
          "activeUsers": "3514",
          "currentUsers": Array [
            "1",
          ],
          "description": "Learn React, an open-source JavaScript library, for building fast webapps and clever User interfaces.",
          "header": "jsimage.jpg",
          "icon": "js.png",
          "id": "1",
          "language": "JavaScript",
          "title": "ReactJS",
        },
        Object {
          "activeUsers": "38",
          "currentUsers": Array [
            "1",
            "2",
          ],
          "description": "Produce concise lambda functions to enhance your functional programming skills. It is recommended that users complete our interfaces lesson first.",
          "header": "javaheader.jpg",
          "icon": "java.png",
          "id": "2",
          "language": "Java",
          "title": "Lambda Functions",
        },
        Object {
          "activeUsers": "802",
          "currentUsers": Array [
            "0",
          ],
          "description": "Learn recursive functions to ace all of your coding interviews. We recommend that all users complete our loops and conditionals lesson before beginning here.",
          "header": "python image.jpg",
          "icon": "python.png",
          "id": "3",
          "language": "Python",
          "title": "Recursion",
        },
        Object {
          "activeUsers": "604",
          "currentUsers": Array [
            "1",
            "2",
          ],
          "description": "Meaning to take many forms, polymorphism is among the four pillars of Object-oriented programming; a must know for all programmers.",
          "header": "javaheader.jpg",
          "icon": "java.png",
          "id": "4",
          "language": "Java",
          "title": "Polymorphism",
        },
        Object {
          "activeUsers": "190",
          "currentUsers": Array [
            "0",
          ],
          "description": "Learn how to write a program that is able to do hundreds of hours of human cognition in seconds through repetition and scripted decision making",
          "header": "python image.jpg",
          "icon": "python.png",
          "id": "5",
          "language": "Python",
          "title": "Control Flow and Conditionals",
        },
        Object {
          "activeUsers": "334",
          "currentUsers": Array [
            "0",
          ],
          "description": "Learn the fundamental elements that make up the world of Object-Oriented Programming.",
          "header": "python image.jpg",
          "icon": "python.png",
          "id": "6",
          "language": "Python",
          "title": "Classes and Objects",
        },
        Object {
          "activeUsers": "2154",
          "currentUsers": Array [
            "0",
          ],
          "description": "Learn arrow functions - introduced with the ECMAScript 6 - and discover how to write shorter, more concise functions with more capable than before.",
          "header": "jsimage.jpg",
          "icon": "js.png",
          "id": "7",
          "language": "JavaScript",
          "title": "ES6 Arrow Functions",
        },
      ]
    }
  />
  <PopularCourses
    courses={
      Array [
        Object {
          "activeUsers": "3514",
          "currentUsers": Array [
            "1",
          ],
          "description": "Learn React, an open-source JavaScript library, for building fast webapps and clever User interfaces.",
          "header": "jsimage.jpg",
          "icon": "js.png",
          "id": "1",
          "language": "JavaScript",
          "title": "ReactJS",
        },
        Object {
          "activeUsers": "38",
          "currentUsers": Array [
            "1",
            "2",
          ],
          "description": "Produce concise lambda functions to enhance your functional programming skills. It is recommended that users complete our interfaces lesson first.",
          "header": "javaheader.jpg",
          "icon": "java.png",
          "id": "2",
          "language": "Java",
          "title": "Lambda Functions",
        },
        Object {
          "activeUsers": "802",
          "currentUsers": Array [
            "0",
          ],
          "description": "Learn recursive functions to ace all of your coding interviews. We recommend that all users complete our loops and conditionals lesson before beginning here.",
          "header": "python image.jpg",
          "icon": "python.png",
          "id": "3",
          "language": "Python",
          "title": "Recursion",
        },
        Object {
          "activeUsers": "604",
          "currentUsers": Array [
            "1",
            "2",
          ],
          "description": "Meaning to take many forms, polymorphism is among the four pillars of Object-oriented programming; a must know for all programmers.",
          "header": "javaheader.jpg",
          "icon": "java.png",
          "id": "4",
          "language": "Java",
          "title": "Polymorphism",
        },
        Object {
          "activeUsers": "190",
          "currentUsers": Array [
            "0",
          ],
          "description": "Learn how to write a program that is able to do hundreds of hours of human cognition in seconds through repetition and scripted decision making",
          "header": "python image.jpg",
          "icon": "python.png",
          "id": "5",
          "language": "Python",
          "title": "Control Flow and Conditionals",
        },
        Object {
          "activeUsers": "334",
          "currentUsers": Array [
            "0",
          ],
          "description": "Learn the fundamental elements that make up the world of Object-Oriented Programming.",
          "header": "python image.jpg",
          "icon": "python.png",
          "id": "6",
          "language": "Python",
          "title": "Classes and Objects",
        },
        Object {
          "activeUsers": "2154",
          "currentUsers": Array [
            "0",
          ],
          "description": "Learn arrow functions - introduced with the ECMAScript 6 - and discover how to write shorter, more concise functions with more capable than before.",
          "header": "jsimage.jpg",
          "icon": "js.png",
          "id": "7",
          "language": "JavaScript",
          "title": "ES6 Arrow Functions",
        },
      ]
    }
  />
</styled.div>
`;
